### PR TITLE
Update dependencies on top of 0.2.0

### PR DESCRIPTION
**Pull request by redcatbear**
#94: Updated Lua dependencies on top of 0.2.0

### CODE REVIEW

File: .busted [hunk 1]
```diff
@@ -28,9 +28,11 @@ return {
     ROOT = {"spec/integration/smoketest_spec.lua"},
   },
   utest = {
+    coverage = true,
     ROOT = {"spec/unit"},
   },
   itest = {
+    coverate = true,
```
Comment (kaklakariada) [inline .busted:35]:
```suggestion
    coverage = true,
```

File: luasql-exasol-0.2.1-1.rockspec [hunk 2]
```diff
@@ -31,7 +31,8 @@ dependencies = {
     "luasocket >= 3.1.0-1",
     "luasec >= 1.3.2-1",
     "luaossl >= 20220711-0",
-    "lua-cjson == 2.1.0", -- pinned to prevent "undefined symbol: lua_objlen" in 2.1.0.6 (https://github.com/mpx/lua-cjson/issues/56)
+    -- "lua-cjson == 2.1.0", -- pinned to prevent "undefined symbol: lua_objlen" in 2.1.0.6 (https://github.com/mpx/lua-cjson/issues/56)
```
Comment (kaklakariada) [inline luasql-exasol-0.2.1-1.rockspec:34]:
```suggestion
```
Is pinning still required?

File: doc/changes/changes_0.2.1.md [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
+# exasol-driver-lua 0.2.1, released 2024-07-??
 
 Code name: Upgrade to Exasol 8
 
 ## Summary
 
-This release updates integration tests to additionally use Exasol version 8.18.1.
+This release updates integration tests to additionally use Exasol version 8.23.0.
+
+When running integration tests with `luarocks test` or `busted`, the environment variable `EXASOL_HOST` now has a default value `localhost` to make running tests more convenient.
```
Comment (Nicoretti) [inline doc/changes/changes_0.2.1.md:9]: Just a side note regarding localhost.
https://www.youtube.com/watch?v=98SYTvNw1kw

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Review by Nicoretti — DISMISSED** [review]

File: doc/changes/changes_0.2.1.md (hunk 3 above)
Comment (redcatbear) [reply doc/changes/changes_0.2.1.md:9]: I would not use that default in production code but in a test scenario, it actually reduces potential failure points.

File: spec/config.lua [hunk 4]
```diff
@@ -21,7 +21,7 @@ end
 
 function M.get_connection_params(override)
     override = override or {}
-    local host = override.host or get_system_env("EXASOL_HOST")
+    local host = override.host or get_optional_system_env("EXASOL_HOST", "localhost")
     local port = override.port or get_system_env("EXASOL_PORT", "8563")
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply spec/config.lua]: Good point. All are optional now in fact, so I was able to further reduce the code.
All defaults are for testing against a vanilla `docker-db`.

File: luasql-exasol-0.2.1-1.rockspec (hunk 2 above)
Comment (redcatbear) [reply luasql-exasol-0.2.1-1.rockspec]: No, the problem seems to be fixed. Removed the info and the commented line.

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**